src/include/cephfs/sharedtypes.h

```diff
@@ -0,0 +1,946 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#ifndef CEPH_SHAREDTYPES_H
```

Comment:
I think this file can be renamed `types.h`. It's already under `cephfs/`.

Reply:
there is already a header named types.h somewhere, won't that be a little mess to have two headers with the same names?

Reply:
IMO it should be okay if there are in different paths.

Reply:
> the ifndef will also be CEPH_TYPES_H, I hope it won't be an issue right?

That definitely will be an issue! Would have to stick the path in there or something.

Reply:
@batrick Can we keep it as sharedypes.h, it also would not lead to ambiguity with the other `types.h` and the name means what it is, i.e. containing sharedtypes?

Reply:
That makes sense, I'll rename it.

Reply:
@batrick I've pushed the changes, PTAL

Reply:
BTW the commits are just to have clarity of what changes I have made so far, I'll squash them up after all the requested changes are incorporated.